### ISSUE TITLE
Drop Tranche `location`

### ISSUE THREAD

**Issue by NunoAlexandre** [centrifuge/centrifuge-chain#1336]
Currently, the `pool_system::tranches::create_metadata` function defines the `AssetMetadata.location` field as follows:

``` rust
location: Some(VersionedMultiLocation::V3(MultiLocation {
	parents: 1,
	interior: X3(
		Parachain(parachain_id.into()),
		PalletInstance(pallet_index),
		// todo(nuno): revisit this
		GeneralKey {
			length: 16u8,
			data: vec_to_fixed_array(tranche_id),
		},
	),
})),
```

I remember suggesting this structure many months back, but as I am [updating the codebase to Polkadot v0.9.38](#1332) and XCM v3 and bumped into this, I realised we don't need a location for `tranche` tokens at all. Tranche tokens are not to be handled through XCM due to their permissions nature and we are building support for permissioned transfers through Connectors.

### Suggestion

Basically set `location` to `None` and adjust all tests.

``` diff
+ location: None,
- location: Some(VersionedMultiLocation::V3(MultiLocation {
- 	parents: 1,
- 	interior: X3(
- 		Parachain(parachain_id.into()),
- 		PalletInstance(pallet_index),
- 		// todo(nuno): revisit this
- 		GeneralKey {
- 			length: 16u8,
- 			data: vec_to_fixed_array(tranche_id),
- 		},
- 	),
- })),
```

**Comment by NunoAlexandre** [comment]
@mustermeiszer @offerijns what do you think? Any objections?

**Comment by mustermeiszer** [comment]
Sounds good and agreed!

**Comment by NunoAlexandre** [comment]
Great, done in https://github.com/centrifuge/centrifuge-chain/pull/1332/commits/61f2960316df62aa743b16ed42f5e4531e236148. I will also include a migration that sets the `location` of any existing tranche token in the registry to `None`.

**Comment by mustermeiszer** [comment]
I am not the biggest fan of having this in 38. But I also know that having intermediate branches is annoying. But could cherru pick the commit and create a PR. Especially given we need a migration. WDYT?

**Comment by NunoAlexandre** [comment]
> I am not the biggest fan of having this in 38. But I also know that having intermediate branches is annoying. But could cherru pick the commit and create a PR. Especially given we need a migration. WDYT?

You are absolutely right. I am always advocating for smaller, self-contained PRs but this time I went against my own principle. I will backport it to a new PR based of the latest main and add the migration part as well. Thanks for the feedback 🤟
